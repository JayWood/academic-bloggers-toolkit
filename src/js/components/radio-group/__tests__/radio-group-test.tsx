import * as React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import RadioGroup from '..';

const setup = () => {
    const onChange = jest.fn();
    const component = shallow(
        <RadioGroup
            name="test"
            label="Test Radio Group"
            value="one"
            items={[
                {
                    label: 'One',
                    value: 'one',
                },
                {
                    label: 'Two',
                    value: 'two',
                },
            ]}
            onChange={onChange}
        />,
    );
    return {
        component,
        onChange,
    };
};

describe('<RadioGroup />', () => {
    it('should match snapshots', () => {
        const { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();
    });
});
