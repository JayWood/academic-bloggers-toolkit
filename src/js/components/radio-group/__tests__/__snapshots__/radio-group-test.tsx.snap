// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> should match snapshots 1`] = `
<div
  aria-labelledby="12345-group"
  className="fieldset"
  role="radiogroup"
>
  <label
    className="legend"
    id="12345-group"
  >
    Test Radio Group
  </label>
  <div
    className="itemContainer"
  >
    <label
      className="item"
      key="12345"
    >
      <input
        aria-checked={true}
        checked={true}
        name="test"
        onChange={[MockFunction]}
        type="radio"
        value="one"
      />
      <span>
        One
      </span>
    </label>
    <label
      className="item"
      key="12346"
    >
      <input
        aria-checked={false}
        checked={false}
        name="test"
        onChange={[MockFunction]}
        type="radio"
        value="two"
      />
      <span>
        Two
      </span>
    </label>
  </div>
</div>
`;
